src/xy.js

```diff
@@ -17,9 +15,15 @@ var XYControl = React.createClass({
   },
   getInitialState() {
     return {
-      dragging: false
+      dragging: false,
+      isMobile: false
     };
   },
+  componentDidMount() {
+    this.setState({
+      isMobile: typeof document != 'undefined' && 'ontouchstart' in document
```

Comment:
I see that we need the `typeof document` check to avoid a crash in a node environment, but is the move to a state variable necessary? If there are, for instance, lots of XYControls on a page, I don't see why each should have its own `isMobile` value, rather than the previous design where this was evaluated once at require time.